src/auth.service.js

```diff
@@ -49,6 +57,19 @@
       user.isAdmin = function() {
         return user.hasRole(options.adminRoles);
       };
+      // add getters for the known properties of an user. This way the
+      // code trying to access user properties can use a getter and
+      // get an exception when trying to access properties which are
+      // unknown to us
+      [
+        'name',
+        'role'
+      ].forEach(function (prop) {
+        var getterName = prop+'Getter';
```

Comment:
Yes, it's a Java naming convention and I thought it was widely used in OOP. You are not forced to use it, this was just a comment.